cylc/flow/workflow_db_mgr.py

```diff
@@ -496,9 +504,23 @@ def put_task_pool(self, pool: 'TaskPool') -> None:
                 }
                 self.db_updates_map.setdefault(self.TABLE_TASK_STATES, [])
                 self.db_updates_map[self.TABLE_TASK_STATES].append(
-                    (set_args, where_args))
+                    (set_args, where_args)
+                )
                 itask.state.time_updated = None
 
+    def put_tasks_to_hold(
+        self, tasks: Set[Tuple[str, 'PointBase']]
+    ) -> None:
+        """Replace the tasks in the tasks_to_hold table."""
+        # As we are replacing the maps each time this is called, there
+        # shouldn't be much cost to calling this multiple times between
+        # processing of the db queue
+        self.db_deletes_map[self.TABLE_TASKS_TO_HOLD] = [{}]
+        self.db_inserts_map[self.TABLE_TASKS_TO_HOLD] = [
+            {"name": name, "cycle": str(point)}
+            for name, point in tasks
+        ]
```

Comment:
Hmm, not sure why I did it this way. There may have been a reason why I didn't only insert/delete the differing tasks, but I can't see it

Reply:
I think this might be the approach used for other tables?

For the moment I've got my fingers in my ears going lalala, the DB usage needs some major rationalisation:

* Repeated wiping and rebuilding of tables.
* Repeated creation and destruction of the DB connection.
* Un-normalised tables.

See https://github.com/cylc/cylc-flow/issues/3872, https://github.com/cylc/cylc-flow/issues/3666.

At the moment it's messy but working and we have bigger things to worry about so I think this work will slide back a while to come.